include/dlr.h

```diff
@@ -47,6 +47,17 @@ typedef void (*DLRFreeFunctionPtr)(void*);
 typedef void* (*DLRMemalignFunctionPtr)(size_t, size_t);
 #endif
 
+#ifndef DLR_MODEL_ELEM
+#define DLR_MODEL_ELEM
+enum DLRModelElemType { HEXAGON_LIB, NEO_METADATA, TVM_GRAPH, TVM_LIB, TVM_PARAMS, RELAY_EXEC };
+typedef struct ModelElem {
+   const DLRModelElemType type;
```

Comment:
should we make this a string (or char*) so that when we have new file types we dont need to update the enum?

Reply:
The common practice is to use string constants or enums for enumerated keys. if we add the support new type then we will add unit tests anyway. Adding new enum element should not create any issues.

Reply:
do we make any claims about API binary compatibility between DLR versions? or do we assume users would recompile their app with the latest dlr.h header file to use a newer version of DLR?

Reply:
It should be possible to use new libdlr.so and older dlr.h in user's applications. Enum is just an integer number. We will add new element to the end of the list.